### PR TITLE
Add test case for binding assets before passing to Definitions

### PR DESCRIPTION
**Pull request by schrockn**
### Summary & Motivation

I thought I was going to have to change code in order to support this use case (binding an asset to with_resources) before passing into Definitions but I was pleasantly surprised to see that this was already supported. Added test case I assumed would have failed.

### How I Tested These Changes

BK

### CODE REVIEW

File: python_modules/dagster/dagster_tests/definitions_tests/test_definitions_class.py [hunk 1]
```diff
@@ -400,3 +401,37 @@ def a_logger(_):
...
+                        "passed-through-with-resources"
+                    )
+                },
+            ),
+            asset_two,
+        ],
+        resources={"b_resource": "passed-through-definitions"},
+    )
+
+    defs.get_job_def("__ASSET_JOB").execute_in_process()
```
Comment (sryza) [inline python_modules/dagster/dagster_tests/definitions_tests/test_definitions_class.py:434]: `get_base_job_for_assets` would let you avoid using the magic string.

**Review by sryza — APPROVED** [review]

**Comment by schrockn** [comment]
This ship has sailed a bit, but I do find the "base job" verbiage to be a bit odd. Base implies a relationship to the *other* jobs in the code location, which is not true. `get_implicit_asset_job` makes more sense to me

**Comment by sryza** [comment]
I don't think the ship has sailed – this isn't a public API. Feel free to change.

**Comment by schrockn** [comment]
will do so in followup

**Pull request merged**